### ISSUE TITLE
Boost bracket_and_solve_root issue: unable to bracket root

### ISSUE THREAD

**Issue by reinhold-willcox** [TeamCOMPAS/COMPAS#317]
**Describe the bug**
Very rarely, the following error occurs.

```
COMPAS v02.13.03
Compact Object Mergers: Population Astrophysics and Statistics
by Team COMPAS (http://compas.science/index.html)
A binary star simulator

Start generating binaries at Thu Aug 13 15:04:13 2020

terminate called after throwing an instance of 'boost::exception_detail::clone_impl<boost::exception_detail::error_info_injector<boost::math::evaluation_error> >'
  what():  Error in function boost::math::tools::bracket_and_solve_root<double>: Unable to bracket root, last nearest value was 3.4657688153706713e-24
Aborted (core dumped)
```

**Label the issue**

`urgency_moderate` - This is a moderately urgent issue

`severity_moderate` - This is a moderately severe bug

**To Reproduce**
On the dev branch, version v02.13.03, run the generic pythonSubmitDefault on the attached grid file.

[bad_grid.txt](https://github.com/TeamCOMPAS/COMPAS/files/5067221/bad_grid.txt)

**Expected behavior**
The binary would complete without the boost error.

**Screenshots**
N/A

**Versioning (please complete the following information):**
 - OS: Ubuntu 18.04 & OzStar
 - COMPAS v2.13.03

**Additional context**
Happens extremely rarely (once in a 1e6 binary run).

**Comment by reinhold-willcox** [comment]
What I notice is that the dM at each iteration is going down. It starts at 0.00205... (our guess), and although factor = 2 and is_rising = 1, the dM drops by half each time (instead of doubling), so boost throws an error when we drop below the system minimum float (or something like that).

**Comment by ilyamandel** [comment]
The solution is simple (catching an exception thrown by the boost root finder).  This, indeed, is very rare.

However, the ultimate reason for the issue is more complex, and has to do with our failing to properly set the core mass of newly minted HG and HeHG stars.  Fixing this has a potentially more significant impact.

Both issues will be addressed by PR #319 .

**Comment by ilyamandel** [comment]
Solved by PR #319 , closing.
